modules/edu.gemini.seqexec.server/src/main/scala/edu/gemini/seqexec/server/GmosEpics.scala

```diff
@@ -256,20 +250,20 @@ object GmosEpics extends EpicsSystem[GmosEpics] {
   override def build(service: CaService, tops: Map[String, String]) = new GmosEpics(service, tops)
 
   case class RoiParameters(cs: Option[CaCommandSender], i: Int) {
-    val ccdXstart = cs.map(_.getInteger(s"ccdXstart$i"))
+    val ccdXstart: Option[CaParameter[Integer]] = cs.map(_.getInteger(s"ccdXstart$i"))
     def setCcdXstart1(v: Integer): SeqAction[Unit] = setParameter(ccdXstart, v)
 
-    val ccdYstart = cs.map(_.getInteger(s"ccdYstart$i"))
+    val ccdYstart: Option[CaParameter[Integer]] = cs.map(_.getInteger(s"ccdYstart$i"))
     def setCcdYstart1(v: Integer): SeqAction[Unit] = setParameter(ccdYstart, v)
 
-    val ccdXsize = cs.map(_.getInteger(s"ccdXsize$i"))
+    val ccdXsize: Option[CaParameter[Integer]] = cs.map(_.getInteger(s"ccdXsize$i"))
     def setCcdXsize1(v: Integer): SeqAction[Unit] = setParameter(ccdXsize, v)
 
-    val ccdYsize = cs.map(_.getInteger(s"ccdYsize$i"))
+    val ccdYsize: Option[CaParameter[Integer]] = cs.map(_.getInteger(s"ccdYsize$i"))
     def setCcdYsize1(v: Integer): SeqAction[Unit] = setParameter(ccdYsize, v)
   }
 
-  class RoiStatus(sa: CaStatusAcceptor, i: Int) {
+  case class RoiStatus(sa: CaStatusAcceptor, i: Int) {
```

Comment:
Just In general I tend to use case classes unless there is a very good reason not to, you get a bunch of goodies, including proper equals and hashCode